comment/api/serializers.py

```diff
@@ -22,10 +24,14 @@ def get_profile_model():
 def get_user_fields():
     user_model = get_user_model()
     fields = user_model._meta.get_fields()
+    api_fields = settings.COMMENT_USER_API_FIELDS
+    api_fields.append('profile')
+    # remove duplicates while preseving order
+    api_fields = list(OrderedDict((x, 1) for x in api_fields))
     for field in fields:
-        if hasattr(field, "upload_to"):
-            return 'id', 'username', 'email', 'profile', field.name
-    return 'id', 'username', 'email', 'profile'
+        if hasattr(field, "upload_to") and isinstance(field, ImageField):
```

Comment:
The first check should be about `ImageField` is what I feel.
```suggestion python
        if isinstance(field, ImageField) and hasattr(field, 'upload_to'):
```